src/ctia/flows/crud.clj

```diff
@@ -1,14 +1,21 @@
 (ns ctia.flows.crud
-  "This namespace handle all necessary flows for creating, updating and deleting objects.
-
-  (Cf. #159)."
+  "This namespace handle all necessary flows for creating, updating and deleting entities."
   (:import java.util.UUID)
   (:require [clojure.tools.logging :as log]
+            [ctia.domain.id :as id]
             [ctia.flows.hooks :as h]
             [ctia.events.obj-to-event :refer [to-create-event
                                               to-update-event
                                               to-delete-event]]))
 
+(defn find-id
+  "Lookup an ID in a given entity.  Parse it, because it might be a
+   URL, and return the short form ID.  Returns nil if the ID could not
+   be found."
+  [{id :id}]
+  (when (not-empty id)
```

Comment:
You are right.  Something to train myself on:  Only use not-empty when I care about the type of the result.